server/src/catalog.rs

```diff
@@ -201,6 +220,52 @@ impl Catalog {
             false
         }
     }
+
+    // TODO: Rename to "try_adding_model_tables" when the old version is removed.
+    /// For each row in the model_table_metadata table, add a model table entry to the catalog. If
+    /// the database could not be opened or the table could not be queried, return [`rusqlite::Error`].
+    fn try_adding_new_model_tables(
+        model_table_metadata: &mut Vec<NewModelTableMetadata>,
+        data_folder_path: &Path
+    ) -> Result<(), rusqlite::Error> {
+        let database_path = data_folder_path.join("metadata.sqlite3");
+        let connection = Connection::open(database_path)?;
+
+        let mut select_statement = connection.prepare("SELECT * FROM model_table_metadata")?;
+        let mut rows = select_statement.query(())?;
+
+        while let Some(row) = rows.next()? {
+            let name = row.get::<usize, String>(0)?;
+
+            // Since the schema is saved as JSON in the database, convert it back to an Arrow schema.
```

Comment:
This was initially done because I was having issues with the conversion to and from bytes and JSON was the easier option. I have changed it to use a BLOB in the table and converting to and from bytes.